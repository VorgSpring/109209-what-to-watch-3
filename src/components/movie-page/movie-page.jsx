import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import {MovieCard} from '../movie-card/movie-card.jsx';
import {MoviesListContainer} from '../movies-list/movies-list.jsx';
import {Footer} from '../footer/footer.jsx';
import {getFilmByIdSelector} from '../../selectors/films/films';
import {RoutePaths} from '../../constants/route-paths.js';

export const MoviePage = ({film}) => {
  // TODO загружать если нету фильма
  if (!film) {
    return <Redirect to={RoutePaths.MAIN} />;
  }

  // TODO брать id из url
  return (
    <Fragment>
      <MovieCard
        film={film}
      />

      <div className='page-content'>
        <section className='catalog catalog--like-this'>
          <h2 className='catalog__title'>More like this</h2>

          <MoviesListContainer genre={film.genre} max={4} excludeId={film.id} />
        </section>

        <Footer />
      </div>
    </Fragment>
  );
};

MoviePage.defaultProps = {
  film: null
};

MoviePage.propTypes = {
  film: PropTypes.object
};

const mapStateToProps = (state, props) => ({
  film: getFilmByIdSelector(state, props)
});

export const MoviePageContainer = connect(
    mapStateToProps
)(MoviePage);
