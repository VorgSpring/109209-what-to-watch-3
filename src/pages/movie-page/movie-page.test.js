import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {MoviePage} from './movie-page.jsx';
import renderer from 'react-test-renderer';

jest.mock(`../../components/movies-list/movies-list.jsx`, () => ({
  MoviesListContainer() {
    return <movie-list />;
  }
}));

jest.mock(`../../components/movie-card/movie-card.jsx`, () => ({
  MovieCard() {
    return <movie-card />;
  }
}));

jest.mock(`../../components/footer/footer.jsx`, () => ({
  Footer() {
    return <footer />;
  }
}));

describe(`MoviePage`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <MoviePage />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
