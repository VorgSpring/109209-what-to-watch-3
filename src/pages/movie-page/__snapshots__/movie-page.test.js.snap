// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should renders correctly 1`] = `null`;
